config/routes.rb

```diff
@@ -64,6 +64,7 @@
 
     resources :application_users, only: [:index] do
       collection do
+        get 'username/:username', action: 'username'
```

Comment:
I either recommended or OK'd this route, but I no longer have a good reaction to it :-)  I think I'd be happier with `/api/application_users/find/:query`, which is like the search route except that it is only for exact matches and `query` has to be something that can achieve an exact match (an ID, a UUID, a username).  

* `.../find/jpslav`
* `.../find/234`
* `.../find/a2ae1a0c-ca60-46c3-926e-cafd674afcf4`

I think we should be able to automatically determine the type of query by the content of the query. If it has hyphens it is a UUID (usernames cannot have hyphens).  If it starts with a number it is an ID (this would require us preventing usernames from starting with numbers --  is this feasible for legacy CNX users?  I'll check with CNX folks).  Otherwise it is a username.

We can't find by email since multiple users can have the same email address.

Thoughts @nathanstitt and @Dantemss?